addons/binding/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/HttpChannelState.java

```diff
@@ -0,0 +1,275 @@
+/*
+ * Copyright (c) 2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.http.internal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.HSBType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.library.types.PointType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.types.UpDownType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.http.model.ErrorListener;
+import org.openhab.binding.http.model.HttpChannelConfig;
+import org.openhab.binding.http.model.Transform;
+import org.openhab.binding.http.util.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.math.BigDecimal;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+
+import static org.openhab.binding.http.HttpBindingConstants.CHANNEL_TYPE_ID_IMAGE;
+import static org.openhab.binding.http.HttpBindingConstants.DEFAULT_CONTENT_TYPE;
+
+/**
+ * This holds the state and handles commands for a channel connected to a HTTP Thing.
+ *
+ * @author Brian J. Tarricone
+ */
+@NonNullByDefault
+public class HttpChannelState implements AutoCloseable {
+    private static State stateFromString(final String stateStr) {
+        final String stateStrTrimmed = stateStr.trim();
```

Comment:
I did the same in MQTT, before someone told me about the `TypeParser` class. It offers static methods to convert a string to a State or Command.